### ISSUE TITLE
Fetch usernames from cs.com

### ISSUE THREAD

**Issue by chmac** [simison/couchspinner#6]
Linked to #5, the main thing I want to do programmatically is grab the CS profile links of all my friends, and those who still have profiles, capture something. Name, maybe picture, but whatever, something more useful than a useless ID.

This would make more sense as part of some kind of one-time export conversion script, rather than on every load of the data.

**Comment by simison** [comment]
Yeah, this would be fantastic! It's a real bummer files don't include at least names.

I actually started implementing it when first working on this, but quickly gave up since it became kinda complicated. There's some code in `add/profile-fetching` branch:

https://github.com/simison/couchspinner/blob/6833f123a1f5fda8a057bd6af2bc8b6915511f9b/src/worker.js

- Since the app runs fully on client-side (it's hosted from Github-pages), you would need to use external service to do the fetching at server-side to overcome CORS issues. It's possible to run something like [CorsAnywhere](https://github.com/Rob--W/cors-anywhere) on Heroku just for this purpose.
- There are mitigations at CS side for this type of crawling tons of profiles. In the code you can see I tried to overcome that with `perimeterx-axios-interceptor`
- Lots of profiles are still private so some of the profiles are still unreachable. Not end of the world of course. :-)
- Related to above, I didn't feel comfortable asking for username+pass, and do the authentication dance. Nobody should enter their credentials on some random website. 😅 Not feeling super strong about this so it would be fine.

Lastly, worth just writing CS folks if they would like to add names to export files. :-D It would be the right thing to do.

**Comment by simison** [comment]
> Lots of profiles are still private so some of the profiles are still unreachable. 

Actually reading the code again, looks like name and image are always available in og-tags even for private profiles. So prolly not an issue. :-)

**Comment by chmac** [comment]
I submitted a PR that does some downloading. It adds a 10s delay between requests. In my ~370 friends, I got 230 real profiles and ~140 copies of the CouchSurfing homepage. I just ran it as a script.

The CORS issue would be a headache to do this client side for sure. 🤔
